test/widgets/message_list_test.dart

```diff
@@ -620,6 +621,93 @@ void main() {
     });
   });
 
+  group('Update Narrow on message move', () {
+    final channel = eg.stream(name: 'move test stream');
+    final otherChannel = eg.stream(name: 'other move test stream');
+    final narrow = TopicNarrow(channel.streamId, 'example topic');
```

Comment:
> A straightforward way to do that is to make a local variable `topic` and use it in all the places that need to be talking about the same topic.

The other easy way that comes to mind is to not specify the topic name at all, but get it from the message with `message.topic` (and/or indirect versions of the same, like `TopicNarrow.ofMessage`).

Reply:
> Probably we should make `eg.streamMessage` generate the topic name at random, to help prevent this pattern, the same way we do stream/channel IDs and message IDs and so on.

Sent #932 for this.

> and none of those assume _what_ the default topic name is, only that it's some constant value, which is also an assumption to avoid making implicit but is a different one.

I ended up deciding this assumption was fine, though; it's convenient in a variety of tests. So I made the randomness happen only once per run, and documented that.